ext/ffi/ffi.c

```diff
@@ -188,6 +188,10 @@ typedef struct _zend_ffi_ctype {
 	zend_ffi_type         *type;
 } zend_ffi_ctype;
 
+/* This is a "mutable" copy of zend_empty_array that prevents asseerts in attempts of iteraton
+ * (see https://github.com/php/php-src/issues/9697) */
+static HashTable _empty_array;
```

Comment:
What about ZTS builds? Technically, having multiple parallel threads calling reset() would be writing to this static array simultaneously.

That'd probably be writing the exact same value concurrently if multiple threads concurrently called reset(), which would be safe on most architectures, but may be unsafe in obscure architectures.

A bigger issue would be whether bugs in opcache, PECLs, the engine, or internal functions could actually write properties to this empty array - those would affect not only the given request, but all subsequent requests.

Doing something like ext/session and putting this in the request globals in rinit would limit the effect of bugs

Reply:
OK. You are right. Lets commit your solution.

> Also, what's the motivation/context for overriding get_properties - if it's just reducing memory for typical workloads, it should be safe to override get_properties_for (which would keep memory low after var_export/debug_zval_dump/var_dump/json_encode)
> Is there anything about FFI specifically that is a reason to avoid populating obj->properties

I think, FFI was developed before ``get_properties_for`` was added. 
obj->properties for FFI objects just don't make sense.
If you like, you may refactor ext/ffi to use `get_properties_for``